tools/src/icon4py/tools/py2fgen/wrappers/diffusion_wrapper.py

```diff
@@ -87,10 +90,10 @@ def diffusion_init(
     nudgecoeff_e: fa.EdgeField[wpfloat],
     rbf_coeff_1: gtx.Field[gtx.Dims[dims.VertexDim, dims.V2EDim], gtx.float64],
     rbf_coeff_2: gtx.Field[gtx.Dims[dims.VertexDim, dims.V2EDim], gtx.float64],
-    mask_hdiff: fa.CellKField[bool],
-    zd_diffcoef: fa.CellKField[wpfloat],
-    zd_vertoffset: gtx.Field[gtx.Dims[dims.CellDim, dims.E2CDim, dims.KDim], gtx.int32],
-    zd_intcoef: gtx.Field[gtx.Dims[dims.CellDim, dims.E2CDim, dims.KDim], gtx.float64],
+    mask_hdiff: Optional[fa.CellKField[bool]],
+    zd_diffcoef: Optional[fa.CellKField[wpfloat]],
+    zd_vertoffset: Optional[gtx.Field[gtx.Dims[dims.CellDim, dims.C2E2CDim, dims.KDim], gtx.int32]],
```

Comment:
Was wrongly using the E2CDim which caused strange bugs because fields might get the wrong dimensions.